backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -736,32 +735,33 @@ private void deleteRoleBindingsOfAccessibleProjects(List<ProjectEntity> allowedP
                 projectId,
                 authService.getVertaIdFromUserInfo(unsignedUser),
                 ModelDBServiceResourceTypes.PROJECT.name());
-        RoleBinding publicReadRoleBinding =
-            roleService.getRoleBindingByName(publicReadRoleBindingName);
-        if (publicReadRoleBinding != null && !publicReadRoleBinding.getId().isEmpty()) {
-          roleService.deleteRoleBinding(publicReadRoleBinding.getId());
+        if (publicReadRoleBindingName != null) {
+          roleBindingNames.add(publicReadRoleBindingName);
         }
       }
 
       // Remove all project collaborators
-      roleService.removeResourceRoleBindings(
-          projectId, project.getOwner(), ModelDBServiceResourceTypes.PROJECT);
+      roleBindingNames.addAll(
+          roleService.getResourceRoleBindings(
+              projectId, project.getOwner(), ModelDBServiceResourceTypes.PROJECT));
 
       // Delete workspace based roleBindings
-      deleteWorkspaceRoleBindings(
-          project.getWorkspace(),
-          WorkspaceType.forNumber(project.getWorkspace_type()),
-          project.getId(),
-          ProjectVisibility.forNumber(project.getProject_visibility()));
+      List<String> workspaceRoleBindingNames =
```

Comment:
Can you also add this batch delete logic for datasets and repositories? (they have almost the same same code. For datasets it's a little different).

Reply:
and for create also

Reply:
Yes it is in my taks list and i will do it later in seperate PR

Reply:
track it in jira please